webapp/channels/src/components/emoji_picker/emoji_picker_overlay.tsx

```diff
@@ -83,12 +85,13 @@ export default class EmojiPickerOverlay extends React.PureComponent<Props> {
 
         const calculatedRightOffset = typeof rightOffset === 'undefined' ? this.emojiPickerPosition(target(), show) : rightOffset;
         const placement = this.getPlacement(target(), spaceRequiredAbove, spaceRequiredBelow, defaultHorizontalPosition, show);
+        const isNotMobileView = !getIsMobileView(store.getState());
```

Comment:
Thanks for redoing that! This looks mostly good, but we should access Redux in a different way.

We generally want to avoid accessing `stores/redux_store` directly in components because changes to the Redux state won't cause the app to re-render if we do that. Since this is a class component, we'd typically access the Redux state using React Redux's `connect` function which makes a "connected component" that simply takes data from Redux and passes it into an unconnected component like this one. `connect` takes a `mapStateToProps` function which gets called with the Redux `state` as a parameter and returns any values that should be passed as props into the component, so you'll need `getIsMobileView` in there.

Since this component doesn't have a connected component yet, you'll want to move this file to `components/emoji_picker/emoji_picker_overlay/emoji_picker_overlay.tsx` and then create a new file called `components/emoji_picker/emoji_picker_overlay/index.ts` to hold the connected component. If you look at similar files such as `components/emoji_picker/index.ts`, you should be able to see how that looks.

Hopefully that makes sense. Feel free to let us know if you have any questions

Reply:
Done.